### PR TITLE
Script updates

### LINKED ISSUE #42: cd to server directory
This is the best server launch script for bash that I have seen. I have one suggestion though that would have saved an idiot like me a lot of time in setting it up on MacOS: include a note in the script or the ReadMe that you need to cd into the server directory before you run "bash ServerStart.sh". Instead of doing what I did which was running "bash /Users/<Username>/server directory/StartServer.sh" which returns: "settings.cfg not found" and "forge installer did not download". 
I'm sure this is super basic, but would probably help a few people who barely know what they are doing.
Thanks for the awesome script!

### PR DESCRIPTION
**Pull request by Dijkstra1**
Many changes and new features closes #42, #26 and #18

### CODE REVIEW

File: ServerStart.sh [hunk 1]
```diff
@@ -20,293 +20,446 @@
...
+    echo "INFO: Starting Server at $(date -u +%Y-%m-%d_%H:%M:%S)" >>logs/serverstart.log 2>&1
+    "$MC_SERVER_JAVA" -Xmx"$MC_SERVER_MAX_RAM" "$MC_SERVER_ARGS" -jar "$MC_SERVER_FORGE_JAR" "$MC_SERVER_GAME_ARGS"
+    b=$?
+    b=1
+    if [[ $b == 0 ]]; then
+        echo "DEBUG: Server ended with code 0" >>logs/serverstart.log
+        a=0
+    else
+        now=$((SECONDS))
+        diff=$($now-$last_crash)
```
Comment (Lartza) [inline ServerStart.sh]: This should be `diff=$now-$last_crash` my mistake

File: ServerStart.sh [hunk 2]
```diff
@@ -20,293 +20,446 @@
...
+    else
+        installer=0
+    fi
+    
+    if [[ $installer == 0 ]]; then
+        echo "DEBUG: no forge installer for MCVER: $MC_SERVER_MCVER and FORGEVER: $MC_SERVER_FORGEVER found"  >>logs/serverstart.log
+        echo "No installer found will try to download it"
+        
+        if [ "${MC_SERVER_FORGEURL}" = "DISABLE" ]; then
+            export MC_SERVER_URL="http://files.minecraftforge.net/maven/net/minecraftforge/forge/${MC_SERVER_MCVER}-${MC_SERVER_FORGEVER}/forge-${MC_SERVER_MCVER}-${MC_SERVER_FORGEVER}-installer.jar"
```
Comment (Lartza) [inline ServerStart.sh]: https?

File: ServerStart.sh [hunk 3]
```diff
@@ -20,293 +20,446 @@
...
+        diff=$($now-$last_crash)
+        if [[ $diff -gt $MC_SERVER_CRASH_TIMER ]]; then
+            a=1
+        else
+            a=$((a+1))
+        fi
+        last_crash=$((SECONDS))
+    fi
+    if [[ "$a" == "$MC_SERVER_CRASH_COUNT" ]]; then
+        echo "The server has crashed to many times"
```
Comment (Lartza) [inline ServerStart.sh]: Typo

**Comment by FallenEagle1** [comment]
Next commits will deal with typos.

File: ServerStart.sh (hunk 2 above)
Comment (Dijkstra1) [reply ServerStart.sh]: It's always been http that is a good idea

**Pull request merged**